New package: py-pydmd

a-saitoh-fj opened the pull request:
New package: py-pydmd

a-saitoh-fj committed:
New package: py-pydmd

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,61 @@
+    # https://github.com/mathLab/PyDMD/pull/133
+    patch('isuue-133.patch', when='@0.3')
+
+    conflicts(
+        '^py-sphinx-rtd-theme@0.5.0',
+        msg='py-sphinx-rtd-theme@0.5.0 fails to build (see https://github.com/spack/spack/issues/19310); '
+        'specify an older version, e.g., py-sphinx-rtd-theme@0.4.3',
+    )
```
Please remove this. If there's a bug with a dependent package, we can fix that package. This package builds just fine if `py-sphinx-rtd-theme@0.5.0` is built. It builds fine on my laptop for example.

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,61 @@
+    depends_on('py-matplotlib', type=('build', 'run'))
+    depends_on('py-future', type=('build', 'run'))
+
+    depends_on('py-nose', type=('build', 'test'))
+    depends_on('py-sphinx@1.4.5', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-rtd-theme', type=('build', 'run'), when='+docs')
```
```suggestion
    depends_on('py-nose', type='test')
    depends_on('py-sphinx@1.4.0:1.4.999', type='build', when='+docs')
    depends_on('py-sphinx-rtd-theme', type='build', when='+docs')
```

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,61 @@
+        'specify an older version, e.g., py-sphinx-rtd-theme@0.4.3',
+    )
+
+    phases = ['build', 'install']
```
This is already the default, you can remove this

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,61 @@
+
+    phases = ['build', 'install']
+
+    def build(self, spec, prefix):
+        self.setup_py('build')
+
+        if '+docs' in spec:
+            with working_dir('docs'):
+                make('html')
+
+    def install(self, spec, prefix):
+        self.setup_py('install')
+
+        if '+docs' in spec:
+            install_tree('docs', prefix.docs)
```
I would do this as:
```python
@run_after('build'):
def build_docs(self):
    if '+docs' in self.spec:
        with working_dir('docs'):
            make('html')

@run_after('install')
def install_docs(self):
    if '+docs' in self.spec:
        install_tree('docs', self.prefix.docs)
```
Then you can get the default install args used.

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,61 @@
+        if '+docs' in spec:
+            install_tree('docs', prefix.docs)
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check_build(self):
+        python('test.py')
```
```suggestion
    def test(self):
        python('test.py')
```
This will override the default `python setup.py test` call.

a-saitoh-fj commented:
Thank you for your advice.
I will check and fix these and then request again.
Please give me some time.

a-saitoh-fj committed:
modified:package.py

a-saitoh-fj commented:
Fixed the problem that was pointed out.

-Removed conflicts processing for py-sphinx-rtd-theme@0.5.0.
-Modify the type set in depends_on.
-Deleted the description of phases.
-Correct the name of the function that installs the document.
-Correct the name of the function that executes the test.

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,49 @@
+    homepage = "https://mathlab.github.io/PyDMD/"
+    url      = "https://github.com/mathLab/PyDMD/archive/v0.3.tar.gz"
+
+    import_modules = ['pydmd']
```
Now that #20023 has been merged, `import_modules` should be auto-detected. You may not need this line anymore.

adamjstewart commented on var/spack/repos/builtin/packages/py-pydmd/package.py:
```diff
@@ -0,0 +1,49 @@
+        if '+docs' in self.spec:
+            install_tree('docs', self.prefix.docs)
+
+    def build_test(self):
+        python('test.py')
```
Now that #20023 has been merged, `PythonPackage` no longer has a test method. You can either remove this and the test dependencies, or add it explicitly:
```suggestion
    @run_after('build')
    @on_package_attributes(run_tests=True)
    def build_test(self):
        python('test.py')
```

a-saitoh-fj committed:
modified: package.py

a-saitoh-fj commented:
Fixed the issue that was pointed out.
- Since "import_modules" is automatically detected by merging #20023, this description has been deleted.
- Since the test method of PythonPackage has disappeared due to the merge of #20023, I added the test explicitly.
